hw/dv/tools/dvsim/sim.mk

```diff
@@ -70,26 +67,40 @@ ifneq (${sw_images},)
 	# test> followed by <index>. The <flag> is considered optional.
 	set -e; \
 	mkdir -p ${sw_build_dir}; \
-	${LOCK_SW_BUILD_DIR} "cd ${proj_root} && \
-		env BUILD_ROOT=${sw_build_dir} ${proj_root}/meson_init.sh"; \
 	for sw_image in ${sw_images}; do \
-		image=`echo $$sw_image | cut -d: -f 1`;  \
+		image=`echo $$sw_image | cut -d: -f 1`; \
 		index=`echo $$sw_image | cut -d: -f 2`; \
 		flags=(`echo $$sw_image | cut -d: -f 3- --output-delimiter " "`); \
+		target_dir=`dirname ${sw_build_dir}/build-bin/$$image`; \
+		mkdir -p $$target_dir; \
+		cd ${proj_root}; \
 		if [[ -z $$image ]]; then \
 			echo "ERROR: SW image \"$$sw_image\" is malformed."; \
 			echo "Expected format: path-to-sw-test:index:optional-flags."; \
 			exit 1; \
 		fi; \
 		if [[ $${flags[@]} =~ "prebuilt" ]]; then \
 			echo "SW image \"$$image\" is prebuilt - copying sources."; \
-			target_dir=`dirname ${sw_build_dir}/build-bin/$$image`; \
-			mkdir -p $$target_dir; \
 			cp ${proj_root}/$$image* $$target_dir/.; \
 		else \
 			echo "Building SW image \"$$image\"."; \
-			target="$$image""_export_${sw_build_device}"; \
-			${LOCK_SW_BUILD_DIR} "ninja -C ${sw_build_dir}/build-out $$target"; \
+			bazel_label="//`dirname $${image}`:`basename $${image}`"; \
+			if [[ $$index == "1" ]]; then \
+				bazel_label+="_sim_dv"; \
+			fi; \
+			bazel_opts="--define DISABLE_VERILATOR_BUILD=true"; \
+			if [[ -z $${BAZEL_PYTHON_WHEELS_REPO} ]]; then \
+				echo "Building \"$${bazel_label}\" on network connected machine."; \
+				bazel_cmd="./bazelisk.sh"; \
+			else \
+				echo "Building \"$${bazel_label}\" on air-gapped machine."; \
+				bazel_opts+=" --distdir=$${BAZEL_DISTDIR} --repository_cache=$${BAZEL_CACHE}"; \
+				bazel_cmd="bazel"; \
+			fi; \
+			$${bazel_cmd} build $${bazel_opts} $${bazel_label}; \
+			find -L $$($${bazel_cmd} info output_path)/ \
+				-type f -name "$$(basename $${image})*" | \
+				xargs -I % cp % $${target_dir}; \
```

Comment:
This doesn't apply in our case because the paths are all well formed (ie: no spaces in path names), so I'll just offer this up as a tip:

You can use `-print0` and `-0` to as args to `find` and `xargs` respectively to deal with badly formed pathnames:
```
find . -name "whatever*" -print0 | xargs -0 my_favorate_command
```

Reply:
Oh neat, thanks!